app/models/clothing_item.rb

```diff
@@ -77,12 +77,14 @@ def set_canonical_clothing_item
     self.unit_weight = canonical_clothing_item.unit_weight
     self.magical_effects = canonical_clothing_item.magical_effects
 
-    set_enchantments if persisted?
+    set_enchantments if persisted? && canonical_clothing_item_id_changed?
   end
 
   def set_enchantments
     return if canonical_clothing_item.enchantments.empty?
 
+    enchantables_enchantments.added_automatically.find_each(&:destroy!)
+
     canonical_clothing_item.enchantables_enchantments.each do |model|
       enchantables_enchantments.find_or_create_by!(
```

Comment:
We might not even need to use `find_or_create_by!` given that this will only run if the canonical model has changed. Let's give it a try locally and see what tests break 🤠

Reply:
Update: tests did break, and the reason they did is that the user-added enchantment, _of course,_ could and probably will be one of the same ones already present on a canonical. So never mind on this one.